lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -341,7 +343,7 @@ export function usePaginatedAPIFetch<
 
   const fetchResult = useAPIFetch<FetchType>(
     nextPageURL ?? path,
-    !nextPageURL ? params : undefined,
+    !nextPageURL ? { ...params, limit: `${ITEMS_PER_PAGE}` } : undefined,
```

Comment:
Rather than fixing the number of items per page, the caller could be revised to check whether there is a next page and if so, treat the response size as a lower-bound. This way if the backend decides to change the default page size, everything will continue to work.

Reply:
Yes, that was actually the way I was planning to implement this in the final version.